### ISSUE TITLE
Define Core/File & Core/Image responsibilities (REMINDER ISSUE)

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#23658]
This is a reminder issue to improve the code and perception of Core/File and Core/Image.
Currently we have these directories mixed up, without clear responsibilities and conventions.
We should investigate and decide how we control them and maybe clear up some conventions.
For example, Core/File should be responsible for any system-file related stuff, meaning that Image would also reuse some parts of it. The confusion comes with inheritance, especially exceptions. What exception should ImageFileNotFound extend (possibly the one in file FileNotFound), same for CannotUnlinkImage & CannotUnlinkFile. If there are any resizing exceptions, then it should extend ImageException I suppose.) etc..

~This should be done until 1.7.8 release, because big part of those directories contents are introduced (so it would be nice to have it clean and avoid BC breaks later)~

EDIT by @matks: Postponed for later, due to not enough time 😞

**Comment by Hlavtox** [comment]
Ping @jolelievre @tleon @nicosomb Guys, should we close this? Relevant?

**Comment by boo-code** [comment]
Had a look at the current state to help answer "is this still relevant?".

The responsibility split is actually mostly coherent already: `Core\File` owns filesystem concerns, `Core\Image` owns image-processing concerns and **reuses** `Core\File` for the filesystem-flavored parts. The convention the issue asks for ("an image file-operation error should extend its `Core\File` counterpart") even exists in code:

- `CannotUnlinkImageException extends CannotUnlinkFileException` ✅ (file-rooted, correct)
- image-processing errors correctly sit under `ImageException`: `ImageOptimizationException`, `ImageFormatConfigurationException` ✅

So it's not undefined — it's **applied inconsistently**. Two concrete deviations remain:

1. **`ImageFileNotFoundException extends ImageException`** — but its direct sibling `CannotUnlinkImageException` extends `CannotUnlinkFileException`. For consistency a "file not found" image error should extend `FileNotFoundException` (`Core\File`), the same way the unlink one does.
2. **`AvifUnavailableException extends \Exception`** — it bypasses `CoreException` entirely (not even under `ImageException`), so it escapes the whole PrestaShop exception hierarchy. This one looks simply wrong regardless of the File/Image debate.

**BC caveat** (why this can't just be patched on a minor): both `FileException` and `ImageException` extend `CoreException`, so anything doing `catch (CoreException)` is unaffected. But re-parenting `ImageFileNotFoundException` from `ImageException` to `FileNotFoundException` drops it out of `catch (ImageException)` blocks — an API contract change, so it belongs in a major per [ADR 0017](https://github.com/PrestaShop/ADR/blob/master/0017-backward-compatibility-promise.md). (`AvifUnavailableException → ImageException` only *adds* `CoreException` ancestry it currently lacks, so it's lower-risk, but still a catch-surface change.)

So: relevant, but minor and BC-constrained. Two reasonable ways forward — either (a) fix the two deviations in the next major and add a one-line convention note ("a filesystem-flavored `Core\Image` exception extends its `Core\File` counterpart; pure image-processing exceptions extend `ImageException`"), since the convention is ~80% in place already and documenting it prevents future drift; or (b) close it and revisit when a major refactor of this area is actually on the table. Happy to put up the major-targeted PR for option (a) if you decide it's worth keeping.
